Xamarin.PropertyEditing.Mac/Controls/Custom/PropertyButton.cs

```diff
@@ -119,6 +135,9 @@ private void PopUpContextMenu ()
 				});
 			}
 
+			if (this.viewModel.SupportsBindings)
+				this.viewModel.CreateBindingRequested += OnBindingRequested;
```

Comment:
This should be happening when the view model is assigned, not when the popup is opened. Because you assign this here and only unassign it when the binding is requested, each time you open the popup before actually hitting create binding, you're assuring that an additional binding dialog will appear.